src/Mod/Part/App/WireJoiner.cpp

```diff
@@ -0,0 +1,2440 @@
+                            << ": " << mkEdge.Error());
+                }
+            }
+            if (splitted.size() <= 1) {
+                ++it;
+                continue;
+            }
+
+            showShape(info.edge, "remove");
+            auto removedEdge = info.edge;
+            it = remove(it);
+            for (const auto &v : splitted) {
+                if (!add(v.edge, false, v.bbox, it))
+                    continue;
+                auto &newInfo = *it++;
+                aHistory->AddModified(v.intersectShape, newInfo.edge);
+                // if (v.intersectShape != removedEdge)
+                //     aHistory->AddModified(removedEdge, newInfo.edge);
+                showShape(newInfo.edge, "split");
+            }
+        }
+    }
+
+    void findSuperEdges()
+    {
+        std::unique_ptr<Base::SequencerLauncher> seq(
+                new Base::SequencerLauncher("Combining edges", edges.size()));
+
+        std::deque<VertexInfo> vertices;
+
+        ++iteration;
+
+        // Join edges (let's call it super edge) that are connected to only one
+        // other edges (count == 2 counts this and the other edge) on one of
+        // its vertices to save traverse time.
+        for (auto it=edges.begin(); it!=edges.end(); ++it) {
+            seq->next(true);
+            auto &info = *it;
+            if (info.iteration == iteration || info.iteration < 0)
+                continue;
+            info.iteration = iteration;
+            // showShape(&info, "scheck");
+
+            vertices.clear();
+            vertices.emplace_back(it, true);
+            edgeSet.clear();
+
+            bool done = false;
+            for (int k=0; k<2; ++k) { // search in both direction
+                auto begin = k==1 ? vertices.back().reversed() : vertices.front();
+                while (true) {
+                    auto currentVertex = k==1 ? vertices.front() : vertices.back();
+                    auto current = currentVertex.edgeInfo();
+                    // showShape(current, "siter", k);
+                    int idx = (currentVertex.start?1:0)^k;
+                    EdgeInfo *found = nullptr;
+                    for (int i=current->iStart[idx]; i<current->iEnd[idx]; ++i) {
+                        const auto &v = adjacentList[i];
+                        auto next = v.edgeInfo();
+                        if (next->iteration < 0 // skipped
+                                || next == current) // skip self (see how adjacent list is built)
+                            continue;
+                        if (v == begin) {
+                            // closed
+                            done = true;
+                            break;
+                        }
+                        if (found // more than one branch
+                            || edgeSet.contains(next)) // or, self intersect
+                        {
+                            // if (found) {
```

Comment:
Should this still be here?

Reply:
It was commented out with https://github.com/realthunder/FreeCAD/commit/b34e9652b5737daf92db03c9637b591b2d516200.

Don't know why he haven't simply deleted them, maybe a placeholder for future improvements/rework?

Probably I'd leave it as is, it does no harm, but if you prefer I can remove it

Reply:
I'm happy with this in.

Reply:
Hmm.  After our research on the history of this file, I suspect that this is missed cleanups from the downsizing of this when it moved from Path to Part.  So maybe it should go away.

Reply:
Ok, I removed it with https://github.com/FreeCAD/FreeCAD/pull/12535/commits/da6cbf3561b5e205270d3b4719f6fcad6c1039c2, leaving a comment